datatypes/src/primitives/spatial_partition.rs

```diff
@@ -70,12 +70,21 @@ impl SpatialPartition2D {
     /// Create a partition from a bbox by snapping to the next pixel
     /// The resulting partition is not equivalent to the bbox but contains it
     pub fn with_bbox_and_resolution(bbox: BoundingBox2D, resolution: SpatialResolution) -> Self {
+        const EPSILON: f64 = 0.000_001;
+
         let lr = bbox.lower_right();
+
+        // add an epsilon to the lower right coordinate to make sure that all pixels are included
+
+        let epsilon_x = resolution.x * EPSILON;
+        let epsilon_y = resolution.y * EPSILON;
+
         Self {
             upper_left_coordinate: bbox.upper_left(),
+
             lower_right_coordinate: Coordinate2D {
-                x: (lr.x / resolution.x).ceil() * resolution.x,
-                y: (lr.y / resolution.y).ceil() * resolution.y,
+                x: ((lr.x + epsilon_x) / resolution.x).ceil() * resolution.x,
+                y: ((lr.y + epsilon_y) / resolution.y).ceil() * resolution.y,
```

Comment:
shouldn't this bis - eplyson_y?

Reply:
yes. i reworked the whole thing and adapted tests...

[bdce194](https://github.com/geo-engine/geoengine/pull/712/commits/bdce194e9986dca3733883c4931520219fc92aeb)